arrow/translate.py

```diff
@@ -0,0 +1,67 @@
+import base64
+
+
+async def translate(reader, options=None):
+    if options is None:
+        options = {}
+    t = Translator(options)
+    return t.translate(reader)
+
+
+class Translator:
+
+    def __init__(self, options=None):
+        if options is None:
+            options = {}
+        defaults = {'b64-decode-enums': False, 'prefix-object-ids': False}
+        self.options = {**defaults, **options}
+
+    def translate(self, reader):
+        enums = _list_enums(reader.writer_schema)
+        results = []
+        for record in reader:
+            if record['name'] != 'metadata':
+                results.append(self._translate_record(record, enums))
+        print(len(results))
+        return results
+
+    def _translate_record(self, record, enums):
+        entity_type = record['name']
+        name = record['id']
+        operations = []
+        for key in record['object']:
```

Comment:
Yes, I agree with this one. The avro file we get _should_ be using the PFB schema and therefore _should not_ be missing this property (see https://raw.githubusercontent.com/uc-cdis/pypfb/master/doc/metadata.svg?sanitize=true). However, since the schema is bundled with the avro file and is not guaranteed to be at all related to PFB, we can infer an unexpected schema if this property is missing which would be a useful error to report back.

Reply:
The test that I wrote exposed that this is one piece of a larger PFB schema validation piece. I'm going to tackle this in a follow-on PR.